dags/mlflow-multimodel-dag.py

```diff
@@ -0,0 +1,274 @@
+from airflow.decorators import task, dag, task_group
+from airflow.utils.task_group import TaskGroup
+from airflow.providers.google.cloud.hooks.bigquery import BigQueryHook
+
+from datetime import datetime
+
+import logging
+import mlflow
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.model_selection import train_test_split, GridSearchCV
+from sklearn.metrics import classification_report, confusion_matrix, ConfusionMatrixDisplay, roc_curve
+from sklearn.linear_model import LogisticRegression
+import lightgbm as lgb
+
+
+docs = """
+MLFlow:
+Airflow can integrate with tools like MLFlow to streamline the model experimentation process. By using the automation and orchestration of Airflow together with MLflow's core concepts (Tracking, Projects, Models, and Registry) Data Scientists can standardize, share, and iterate over experiments more easily.
+
+
+XCOM Backend:
+By default, Airflow stores all return values in XCom. However, this can introduce complexity, as users then have to consider the size of data they are returning. Futhermore, since XComs are stored in the Airflow database by default, intermediary data is not easily accessible by external systems.
+By using an external XCom backend, users can easily push and pull all intermediary data generated in their DAG in GCS.
+"""
+
+
+def log_roc_curve(y_test: list, y_pred: list):
+    fpr, tpr, thresholds = roc_curve(y_test, y_pred)
+    plt.plot(fpr,tpr) 
+    plt.ylabel('False Positive Rate')
+    plt.xlabel('True Positive Rate')
+    plt.title('ROC Curve')
+    plt.savefig("roc_curve.png")
+    mlflow.log_artifact("roc_curve.png")
+    plt.close()
+
+
+def log_confusion_matrix(y_test: list, y_pred: list):
+    cm = confusion_matrix(y_test, y_pred)
+    t_n, f_p, f_n, t_p = cm.ravel()
+    mlflow.log_metrics({'True Positive': t_p, 'True Negative': t_n, 'False Positive': f_p, 'False Negatives': f_n})
+
+    ConfusionMatrixDisplay.from_predictions(y_test, y_pred)
+    plt.savefig("confusion_matrix.png")
+    mlflow.log_artifact("confusion_matrix.png")
+    plt.close()
+
+
+def log_classification_report(y_test: list, y_pred: list):
+    cr = classification_report(y_test, y_pred, output_dict=True)
+    logging.info(cr)
+    cr_metrics = pd.json_normalize(cr, sep='_').to_dict(orient='records')[0]
+    mlflow.log_metrics(cr_metrics)
+
+
+def log_all_eval_metrics(y_test: list, y_pred: list):
+    
+    # Classification Report
+    log_classification_report(y_test, y_pred)
+
+
+    # Confusion Matrix
+    log_confusion_matrix(y_test, y_pred)
+
+
+    # ROC Curve
+    log_roc_curve(y_test, y_pred)
+
+
+def test(clf, test_set, model_type):    
+    logging.info('Gathering Validation set results')
+    y_pred = clf.predict(test_set)
+
+    if model_type == 'lgbm':
+        return np.where(y_pred > 0.5, 1, 0)
```

Comment:
Instead of doing a string match, you could just apply this to all models regardless of time, because 0 and 1 will themselves evaluate to 0 and 1. And that way it will generalize to all models and we can add more models in the future.